### PR TITLE
Make consistent interfaces for SeedPython and inline text of SeedBlock

### PR DESCRIPTION
**Pull request by codingpotato**
- Replace eval statement with expression statement.
- Remove ParseRule enum, because only two parse rules (`program` and `single statement`) are needed after the first change.
- Adjust the interfaces of Runtime Executor to separate ParseSyntaxTokens() and Run()
- The SeedLang.Shell can run SeedPython and BlockInlineText now. Do we need run BlockInlineText in shell?
- Change InlineTextToBlock method to do conversion only when the input text is valid.

### CODE REVIEW

File: csharp/src/SeedLang/Block/Converter.cs [hunk 1]
```diff
@@ -96,9 +96,9 @@ public class Converter {
         foreach (var rootBlock in module.RootBlockIterator) {
           // TODO: implement a visitor to parse other kinds of blocks.
           if (rootBlock is ExpressionBlock expressionBlock) {
-            var parser = new InlineTextParser();
-            if (parser.Parse(expressionBlock.GetEditableText(), module.Name, ParseRule.Expression,
-                             collection, out AstNode node, out IReadOnlyList<SyntaxToken> _)) {
+            var parser = new BlockInlineTextParser();
```
Comment (wixette) [inline csharp/src/SeedLang/Block/Converter.cs:99]: As for the naming:

* It seems all the functions that convert SeedX source code to syntax tree are named as `ParseXXX` or similar patterns.
* Meanwhile, SeedBlock uses a particular name pattern `Converter.Convert` to do a different but kind of similar task.
* And, the naming in `BlockInlineTextParser.cs` does follow the `ParseXXX` pattern like other SeedX languages.

I am thinking if the name `Convert` is a good fit here, with no preference yet. Just some random thoughts:

* `Convert` is short and clean, but it looks a bit out-of-context compared to other `ParseXXX` names.
* Not sure if `Translate` is a better naming solution...
* Not sure if `Parser...` still fits here...

Anyway, we can keep `Convert` for now and postpone this decision...

File: csharp/src/SeedLang/Ast/Statements.cs [hunk 2]
```diff
@@ -17,15 +17,15 @@
...
+    // The factory method to creates an assignment statement.
     internal static AssignmentStatement Assignment(IdentifierExpression identifier, Expression expr,
                                                    Range range) {
       return new AssignmentStatement(identifier, expr, range);
     }
 
-    // The factory method to creates the eval statement.
-    internal static EvalStatement Eval(Expression expr, Range range) {
-      return new EvalStatement(expr, range);
+    // The factory method to creates an expression statement.
```
[the remark this answers is not in the document]
Comment (codingpotato) [reply csharp/src/SeedLang/Ast/Statements.cs:26]: Done

File: csharp/src/SeedLang/Ast/Statements.cs [hunk 3]
```diff
@@ -17,15 +17,15 @@
 namespace SeedLang.Ast {
   // The base class of all statement nodes.
   internal abstract class Statement : AstNode {
-    // The factory method to creates the assignment statement.
+    // The factory method to creates an assignment statement.
```
[the remark this answers is not in the document]
Comment (codingpotato) [reply csharp/src/SeedLang/Ast/Statements.cs:20]: Done

File: csharp/src/SeedLang/Block/Converter.cs (hunk 1 above)
Comment (codingpotato) [reply csharp/src/SeedLang/Block/Converter.cs:99]: I don't think Converter is good here too. But let's keep it open now. Maybe we can find a better name after we finalize how to handle the workflow of editing and converting inline texts to blocks.

File: csharp/src/SeedLang/Block/Converter.cs [hunk 4]
```diff
@@ -29,25 +29,26 @@ public class Converter {
...
+      var parser = new BlockInlineTextParser();
+      return parser.Validate(text, "", collection);
     }
 
-    // Converts an expression inline text to a list of blocks. A list of invalidTokenRanges can be
-    // passed in, to collect the text ranges of invalid tokens.
-    public static IReadOnlyList<BaseBlock> InlineTextToBlocks(
-        string text, IList<TextRange> invalidTokenRanges, DiagnosticCollection collection) {
+    // Converts an expression inline text to a list of blocks. Returns null if the inline text is
+    // null, empty or invalid.
```
[the remark this answers is not in the document]
Comment (wixette) [reply csharp/src/SeedLang/Block/Converter.cs:37]: This makes sense here.

> call parse() method in SeedLang.X to parse the inline text into syntax tokens list, and use this list to do highlight. It's not implemented now.

Okay, cool, we will get back to this topic once the parse() method is implemented.

**Review by wixette — APPROVED** [review]

**Pull request merged**